libs/settings/tests/unit/setting_collection_tests.cpp

```diff
@@ -0,0 +1,74 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "settings/setting.hpp"
+#include "settings/setting_collection.hpp"
+
+#include "gtest/gtest.h"
+
+#include <memory>
+
+namespace {
+
+using fetch::settings::Setting;
+using fetch::settings::SettingCollection;
+
+class SystemArgAdapter
+{
+public:
+  SystemArgAdapter(std::vector<std::string> args)
+    : args_{std::move(args)}
+  {
+    argv_.resize(args_.size(), nullptr);
+
+    for (std::size_t idx = 0; idx < args_.size(); ++idx)
+    {
+      argv_[idx] = const_cast<char *>(args_[idx].c_str());
+    }
+  }
+
+  int argc()
+  {
+    return static_cast<int>(argv_.size());
+  }
+
+  char **argv()
+  {
+    return argv_.data();
+  }
+
+private:
+  std::vector<std::string> args_;
+  std::vector<char *>      argv_;
+};
+
+TEST(SettingCollectionTests, SimpleCheck)
```

Comment:
This is nice, but while we're here, can we test stuff typical edge cases? Passing in no settings, passing in a misspelled option, passing in `xyz` for the number of lanes?

Reply:
Done